src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -2280,6 +2430,15 @@ public ParameterResource(string name, System.Func<ParameterDefault?, string> cal
         public System.Threading.Tasks.ValueTask<string?> GetValueAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
     }
 
+    public sealed partial class PortAllocator : IPortAllocator
```

Comment:
@davidfowl - does this need to be public? Or only the interface?

Reply:
Used in other libraries. Needs to be public. We could opt to just make a class with virtual methods - like System.Random, but that is not the established way in the rest of the APIs.